### ISSUE TITLE
Changing DisplayedValues can cause a crash.

### ISSUE THREAD

**Issue by VincentSit** [ShawnLin013/NumberPicker#158]
Changing DisplayedValues to a shorter array, and then changing to a longer array, will certainly cause a crash.

Add the following code to the simple to reproduce the bug.

```Java
...
// Set string values
        String[] data = {"A", "B", "C", "D", "E", "F", "G", "H", "I"};
        numberPicker.setMinValue(1);
        numberPicker.setMaxValue(data.length);
        numberPicker.setDisplayedValues(data);

        final Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                String[] data = {"W", "Y", "Z"};
                numberPicker.setMinValue(1);
                numberPicker.setMaxValue(data.length);
                numberPicker.setDisplayedValues(data);

                final Handler handler = new Handler();
                handler.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        String[] data = {"A", "B", "C", "D", "E", "F", "G", "H", "I"};
                        numberPicker.setMinValue(1);
                        numberPicker.setMaxValue(data.length);
                        numberPicker.setDisplayedValues(data);
                    }
                }, 2000);
            }
        }, 2000);
...
```

**Comment by VincentSit** [comment]
It's an array OOB issue. Currently my temporary solution is to set the `DisplayedValues` in advance.

```java
// Set string values
        String[] data = {"A", "B", "C", "D", "E", "F", "G", "H", "I"};
        numberPicker.setMinValue(1);
        numberPicker.setMaxValue(data.length);
        numberPicker.setDisplayedValues(data);

        final Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                String[] data = {"W", "Z", "Y"};
                numberPicker.setMinValue(1);
                numberPicker.setMaxValue(data.length);
                numberPicker.setDisplayedValues(data);

                final Handler handler = new Handler();
                handler.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        String[] data = {"A", "B", "C", "D", "E", "F", "G", "H", "I"};
                        numberPicker.setDisplayedValues(data);  <-- Add this line.
                        numberPicker.setMinValue(1);
                        numberPicker.setMaxValue(data.length);
                        numberPicker.setDisplayedValues(data);
                    }
                }, 2000);
            }
        }, 2000);
```

**Comment by ShawnLin013** [comment]
Fixed at v2.4.9.
